cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/ArtifactHttpHandler.java

```diff
@@ -170,9 +170,14 @@ public void refreshSystemArtifacts(HttpRequest request, HttpResponder responder)
   @POST
   @Path("/namespaces/{namespace-id}/artifactproperties")
   public void getArtifactProperties(FullHttpRequest request, HttpResponder responder,
-                                    @PathParam("namespace-id") String namespaceId) throws Exception {
+                                    @PathParam("namespace-id") String namespaceId,
+                                    @QueryParam("limit") @DefaultValue("2147483647") String limit,
+                                    @QueryParam("order") @DefaultValue("DESC") String order) throws Exception {
 
     NamespaceId namespace = validateAndGetNamespace(namespaceId);
+    int limitNumber = Integer.valueOf(limit);
```

Comment:
I dont see any actual use case for the limit. I added it to be consistent with the get artifact versions api. I think we can set the default value as 1 and keep it here in case someone wants to query more results

Reply:
But there the limit is a limit for the total number. Here it is a weird limit per item

Reply:
hmm, yea.. then I will just remove it to avoid the confusion